rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -504,10 +504,14 @@ def attempt_connect(
 
     def connect_to_multiple_nodes(self, nodes: Sequence[WeightedNode]) -> None:
         self.web3_mapping = {}
-        for weighted_node in nodes:
-            if weighted_node.node_info.name == self.etherscan_node_name:
-                continue
 
+        # remove etherscan nodes
+        nodes = [node for node in nodes if node.node_info.name != self.etherscan_node_name]
+        if not nodes:
+            return
+
+        self.etherscan_block = self.query_highest_block()
+        for weighted_node in nodes:
```

Comment:
as you can see you are looping twice here. I guess you have it to remove the etherscan node and then query the highest block... mmm this should be stated in the code why you do it to avoid any future person from chaging the code and breaking your assumptions but not sure if we can do better

```suggestion
        self.etherscan_block = self.query_highest_block()
        for weighted_node in nodes:
            if weighted_node.node_info.name == self.etherscan_node_name:
                continue
            if len(self.web3_mapping) == 0:
                  self.etherscan_block = self.query_highest_block()
```

It's one extra check that is annoying but you avoid looping twice